agent/mapcall/src/main/java/randoop/instrument/CallReplacementTransformer.java

```diff
@@ -169,310 +95,328 @@ public CallReplacementTransformer() {}
-      return (null);
-    } else if (fullClassName.startsWith("sun.reflect")) {
-      debug_transform.log("ignoring system class %s, in sun.reflect package", fullClassName);
-      return (null);
-    } else if (fullClassName.startsWith("com.sun")) {
-      System.out.printf("Class from com.sun package %s with nonnull loaders\n", fullClassName);
+    if (isNonGUIBootClass(loader, fullClassName)) {
+      return null;
     }
 
-    // Don't intrument our code
-    if (className.startsWith("randoop.")) {
-      debug_transform.log("Not considering randoop class %s%n", fullClassName);
-      return (null);
-    }
+    debug_transform.log("%ntransform: ENTER %s%n", className);
 
-    // Look for match with specified regular expressions for class
-    Map<MethodDef, ReplacementClass> current_method_map = null;
-    debug_class = false;
-    for (MethodMapInfo mmi : map_calls_file) {
-      if (mmi.class_regex.matcher(className).matches()) {
-        if (false && className.startsWith("RandoopTest")) debug_class = true;
-        if (debug_class)
-          System.out.printf("Classname %s matches re %s%n", className, mmi.class_regex);
-        current_method_map = mmi.map;
-        break;
-      }
-    }
-    if (current_method_map == null) {
-      // no replacements to be done in this class
+    if (isExcludedClass(fullClassName)) {
+      debug_transform.log("transform: EXIT ignoring excluded class %s%n", className);
       return null;
     }
 
-    debug_transform.log(
-        "transforming class %s, loader %s - %s%n", className, loader, loader.getParent());
+    if (loader != null) {
+      debug_transform.log(
+          "transforming class %s, loader %s - %s%n", className, loader, loader.getParent());
+    } else {
+      debug_transform.log("transforming class %s, null - null%n", className);
+    }
 
-    // Parse the bytes of the classfile, die on any errors
+    // Parse the bytes of the classfile
     JavaClass c;
     try {
       ClassParser parser = new ClassParser(new ByteArrayInputStream(classfileBuffer), className);
       c = parser.parse();
     } catch (Exception e) {
-      throw new RuntimeException("Unexpected error", e);
+      debug_transform.log("transform: EXIT parse of %s resulted in error %s%n", className, e);
+      throw new RuntimeException("Unexpected parse exception: ", e);
     }
 
     try {
       // Get the class information
       ClassGen cg = new ClassGen(c);
-      ifact = new InstructionFactory(cg);
-
-      map_calls(cg, className, loader, current_method_map);
-
-      JavaClass njc = cg.getJavaClass();
-      if (debug) njc.dump("/tmp/ret/" + njc.getClassName() + ".class");
-
-      if (true) {
-        return (cg.getJavaClass().getBytes());
+      if (transformClass(cg)) {
+        if (MapCallsAgent.debug) {
+          JavaClass njc = cg.getJavaClass();
+          njc.dump("/tmp/ret/" + njc.getClassName() + ".class");
```

Comment:
Please don't hardcode directory names.  If you do, use a longer and more descriptive directory name than "ret", so that people who stumble across the files have a better chance of understanding what they are.